runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -2018,6 +2061,41 @@ MM_IncrementalGenerationalGC::postConcurrentUpdateStatsAndReport(MM_EnvironmentB
 	env->_cycleState = NULL;
 }
 
+void
+MM_IncrementalGenerationalGC::calculateConcurrentMarkWorkTime(MM_EnvironmentBase *env, MM_ConcurrentPhaseStatsBase *stats)
+{
+	/* Get an estimate of total process time now that concurrent mark increment has finished*/
+	omrthread_process_time_t processEnd;
+	omrthread_get_process_times(&processEnd);
+	U_64 processEndTime = processEnd._systemTime + processEnd._userTime;
+
+	/* Calculate how much process time has elapsed since the start of the concurrent mark increment */
+	U_64 concurrentElapsedTime = processEndTime - stats->_concurrentMarkProcessStartTime;
+
+	/* Work time we attribute to concurrent work is difference of process time of gc threads now vs when concurrent phase started */
+	MM_MarkVLHGCStats markStats = _persistentGlobalMarkPhaseState._vlhgcIncrementStats._markStats;
+	double concurrentGCRatio = 0.5;
+
+	if (markStats._concurrentGCThreadsCPUEndTimeSum != markStats._concurrentGCThreadsCPUStartTimeSum) {
+		/* If the platform supports getting time for each thread, calculate what % of time gc threads were active relative to application threads. */
+		/* Determine the ratio of time spent doing GC work vs non-gc related work during the concurrent phase */
+		U_64 concurrentGCWorkTime = markStats._concurrentGCThreadsCPUEndTimeSum - markStats._concurrentGCThreadsCPUStartTimeSum;
+		concurrentGCRatio = (double)concurrentGCWorkTime/concurrentElapsedTime;
```

Comment:
hm, this ratio is a _sum_ of CPU from _multiple_ GC threads vs elapsed time. this is inflated number (by a number of threads). I guess this is typically > 1.0

Reply:
We are comparing the sum of process time for GC threads to elapse total cpu time for all threads (application threads and GC threads). 

`concurrentElapsedTime ` is indeed very misleading. It should in fact be called, `concurrentElapsedCPUTime`, as per our discussion offline.

Renaming `concurrentGCWorkTime` to `concurrentGcCPUWorkTime` might also be a helpful touch for anyone looking at this code later on

Reply:
Naming of these variables for increased clarity has been resolved as per latest commit